merkle/src/store.rs

```diff
@@ -22,6 +24,10 @@ pub struct StoreConfig {
     /// location for this particular data.
     pub id: String,
 
+    /// The number of elements in the DiskStore.  This field is
+    /// optional, and unused internally.
+    pub size: usize,
```

Comment:
maybe make it Option<usize> if it is optional?

Reply:
what is the purpose of this, if it is not used internally?

Reply:
It can be set externally (this was a convenient place to stash this info while integrating into seal process).  In short, we can track StoreConfig objects instead of DiskStores.  But when we need to access the DiskStore we need to know the size, which could be tucked away inside the config at the time we're storing it for tracking.  Not opposed to making it an option.

Reply:
For context, I'm working with this on the other side:

```
#[derive(Debug)]
pub struct Encodings<H: Hasher> {
    encodings: Vec<StoreConfig>,
    _h: PhantomData<H>,
}

...

#[derive(Debug)]
pub struct TemporaryAux<H: Hasher, G: Hasher> {
    /// The encoded nodes for 1..layers.                                                                                                                                
    pub encodings: Encodings<H>,
    pub tree_d_config: StoreConfig,
    pub tree_r_last_config: StoreConfig,
    pub tree_c_config: StoreConfig,
    pub _g: PhantomData<G>,
}
```

Reply:
Also, the only reason we need the size when loading a disk store from disk is to assert that what we're loading on disk matches the number of elements that we're expecting to have in the store.  Technically, that kind of checking could go away, but seems safer to keep.  So it's specified at the time of load from disk.